vault/quotas/quotas.go

```diff
@@ -167,13 +168,13 @@ type Manager struct {
 	metricSink *metricsutil.ClusterMetricSink
 
 	// quotaLock is a lock for manipulating quotas and anything not covered by a more specific lock
-	quotaLock *sync.RWMutex
+	quotaLock *locking.DeadlockRWMutex
```

Comment:
For my own education: if these `locking.DeadlockRWMutex` are better at detecting deadlocks, what situations should we prefer `sync.RWMutex` over them, if any?

Reply:
They add overhead, so we don't actually use them unless compiled with a special build tag (`deadlock`).

Reply:
I see - thanks!

Reply:
I guess the followup question would be: why not use them everywhere, since there's no cost with our default build, and if we're making a build to catch deadlocks, why limit ourselves?  Maybe we should, I don't know, worth further discussion.